provider/server/server_suite_test.go

```diff
@@ -4,31 +4,40 @@
 package server_test
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
 	"testing"
 	"time"
 
-	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
-	"github.com/ironcore-dev/libvirt-provider/pkg/api"
-	"github.com/ironcore-dev/libvirt-provider/pkg/host"
-	"github.com/ironcore-dev/libvirt-provider/pkg/mcr"
-	"github.com/ironcore-dev/libvirt-provider/pkg/utils"
-	"github.com/ironcore-dev/libvirt-provider/provider/server"
+	"github.com/digitalocean/go-libvirt"
+	"github.com/digitalocean/go-libvirt/socket/dialers"
+	iriv1alpha1 "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	"github.com/ironcore-dev/ironcore/iri/remote/machine"
+	"github.com/ironcore-dev/libvirt-provider/provider/cmd/app"
+	"github.com/ironcore-dev/libvirt-provider/provider/networkinterfaceplugin"
 	. "github.com/onsi/ginkgo/v2"
+
 	. "github.com/onsi/gomega"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
 var (
-	srv *server.Server
+	machineClient iriv1alpha1.MachineRuntimeClient
+	libvirtConn   *libvirt.Libvirt
 )
 
 const (
-	eventuallyTimeout    = 3 * time.Second
+	eventuallyTimeout    = 30 * time.Second
 	pollingInterval      = 50 * time.Millisecond
 	consistentlyDuration = 1 * time.Second
 	baseURL              = "http://localhost:8080"
 	machineClassx3xlarge = "x3-xlarge"
+	streamingAddress     = "127.0.0.1:20251"
```

Comment:
static port can be problematic on shared environment because multiple test cannot run in the same time. Are you okay with that?
@hardikdr can be baseURL different as streamingAddress? because you have note in flag:
```golang
fs.StringVar(&o.BaseURL, "base-url", "", "The base url to construct urls for streaming from. If empty it will be "+
		"constructed from the streaming-address")
```

it looks as dependency between baseURL and streamingAddress

Reply:
Noted. Currently, the static port is not causing any issues, but we'll reassess if it causes any trouble with more tests.